File: IN1000/Obliger/Uke 10 og 11 vanlig redo/rutenett.py
from random import randint
from celle import Celle

class Rutenett:
    def __init__(self, rader, kolonner):
        self._ant_rader = rader
        self._ant_kolonner = kolonner
        self._rutenett = self._lag_tomt_rutenett()

    def _lag_tomt_rutenett(self):
        return [self._lag_tom_rad() for _ in range(self._ant_rader)]

    def _lag_tom_rad(self):
        return [None for _ in range(self._ant_kolonner)]

    def fyll_med_tilfeldige_celler(self):
        for rad in range(self._ant_rader):
            for kol in range(self._ant_kolonner):
                self.lag_celle(rad, kol)

    def lag_celle(self, rad, kol):
        self._rutenett[rad][kol] = Celle()
        if not randint(0,2):
            self._rutenett[rad][kol].sett_levende()

    def hent_celle(self, rad, kol):
        if rad < 0 or rad > self._ant_rader - 1 or kol < 0 or kol > self._ant_kolonner - 1:
            return None
        return self._rutenett[rad][kol]

    def tegn_rutenett(self):
        for _ in range(10):
            print()
        for rad in self._rutenett:
            for celle in rad:
                print(celle, end = " ")
            print()

    def _sett_naboer(self, rad, kol):
        for x in range(-1, 2):
            for y in range(-1, 2):
                if abs(x) + abs(y) != 0:
                    celle = self.hent_celle(rad + x, kol + y)
                    if celle:
                        self._rutenett[rad][kol].legg_til_nabo(celle)

    def koble_celler(self):
        for rad in range(self._ant_rader):
            for kol in range(self._ant_kolonner):
                self._sett_naboer(rad, kol)

    def hent_alle_celler(self):
        return [row for kol in self._rutenett for row in kol]

    def antall_levende(self):
        ant_levende = 0
        for celle in self.hent_alle_celler():
            if celle.er_levende():
                ant_levende += 1
        return ant_levende

    def __str__(self):
        self.tegn_rutenett()
        return ""

File: IN1010/Obliger2/Oblig1/Rutenett.java



public class Rutenett {
    public int antRader;
    public int antKolonner;
    public Celle[][] rutene;

    public Rutenett(int antRader, int antKolonner) {
        this.antRader = antRader;
        this.antKolonner = antKolonner;
        rutene = new Celle[antRader][antKolonner];
    }

    public void lagCelle(int rad, int kol) {
        Celle celle = new Celle();
        if (Math.random() <= 0.3333) celle.settLevende();
        rutene[rad][kol] = celle;
    }

    public void fyllMedTilfeldigeCeller() {
        for (int rad = 0; rad < antRader; rad++) {
            for (int kol = 0; kol < antKolonner; kol++) {
                lagCelle(rad, kol);
            }
        }
    }

    public Celle hentCelle(int rad, int kol) {
        if (rad < 0 || kol < 0 || rad >= antRader || kol >= antKolonner) {
            return null;
        }
        return rutene[rad][kol];
    }

    public void tegnRutenett() {
        for (int rad = 0; rad < antRader; rad++) {
            for (int kol = 0; kol < antKolonner; kol++) {
                System.out.print(rutene[rad][kol].hentStatusTegn());
            }
            System.out.println();
        }
    }

    public void settNaboer(int rad, int kol) {
        Celle celle = rutene[rad][kol];
        for (int r = -1; r < 2; r++) {
            for (int k = -1; k < 2; k++) {
                if (Math.abs(r) + Math.abs(k) != 0) celle.leggTilNabo(hentCelle(rad+r, kol+k));
            }
        }
    }

    public void kobleAlleCeller() {
        for (int rad = 0; rad < antRader; rad++) {
            for (int kol = 0; kol < antKolonner; kol++) {
                settNaboer(rad, kol);
            }
        }
    }

    public int antallLevende() {
        int antLevende = 0;
        for (int rad = 0; rad < antRader; rad++) {
            for (int kol = 0; kol < antKolonner; kol++) {
                if (rutene[rad][kol].erLevende()) antLevende++;
            }
        }
        return antLevende;
    }


}
